base/complex.jl

```diff
@@ -1065,37 +1091,75 @@ julia> div(-5 + 5im, 2 + 2im, RoundNearestTiesUp)
 0 + 3im
 ```
 """
-function div(a::Complex{T}, b::Complex{V}, r::RoundingMode=RoundNearest) where {T<:Integer, V<:Integer}
-    R = promote_type(T, V)
-    a, b = Complex{R}(a), Complex{R}(b)
+function div(a::Complex{T}, b::Complex{T}, rr::RoundingMode=RoundNearest, ri::RoundingMode=rr) where T<:Integer
+    (T != BigInt) && (imag(b) === typemin(T)) && throw(OverflowError("Cannot compute conj($b)."))
```

Comment:
This check for the existence of `typemin` is not reliable (there could be other subtypes of `Integer` not having it). For `T <: Unsigned` the error is thrown wrongly if `imag(b) == 0`.
You would do better with `T <: Base.BitSigned`.